cf-oscore/src/main/java/org/eclipse/californium/oscore/RequestDecryptor.java

```diff
@@ -123,4 +135,38 @@ public static Request decrypt(Request request) throws CoapOSException {
 		request.getOptions().setOscore(rid);
 		return OptionJuggle.setRealCodeRequest(request, ctx.getCoAPCode());
 	}
+
+	/**
+	 * Checks if an incoming messages should trigger re-derivation of the security
+	 * context as detailed in Appendix B.2. If so this re-derivation is also performed.
+	 *
+	 * See https://tools.ietf.org/html/draft-ietf-core-object-security-16#section-5.2
+	 *
+	 * @throws CoapOSException if re-generation of the context fails
+	 */
+	private static void checkContextRederivation(byte[] rid, byte[] receivedContextID) throws CoapOSException {
+		//Get the context corresponding to the incoming rid
+		OSCoreCtxDB db = HashMapCtxDB.getInstance();
+		OSCoreCtx ctx = db.getContext(rid);
+
+		//Check if the received Context ID matches the one in the context, if so do nothing
+		if(receivedContextID == null || Arrays.equals(receivedContextID, ctx.getIdContext())) {
```

Comment:
Please insert a space (' ') between "if" and "(".

Reply:
Fixed in commit c2bffa2b0b1144b65108adbd7f1cc08638928d44